src/python/grpcio/grpc/_channel.py

```diff
@@ -208,7 +221,8 @@ def consume_request_iterator():  # pylint: disable=too-many-branches
                         else:
                             return
                         while True:
-                            state.condition.wait()
+                            state.condition.wait(condition_wait_timeout)
```

Comment:
If fork support is enabled, a generator can't take longer than a second to generate a request?

Reply:
This goes from a wait-until to a wait-one-second-and-then-wait-again loop, where the event being waited for is either of the following to occur: (a) the completion queue returns our send_message operation's tag, or (b) the call has meanwhile been completed (state.code is not None)

The issue is that `state.condition.wait()` may block "forever" waiting for core to complete the send op. This interferes with the book-keeping in the python fork handler, as it will want to wait for this thread to pause or complete. It would actually be fine for the fork to proceed if the parent was still waiting, as the lock is released by the call to `wait()`, but that information would have to be conveyed atomically with the wait call. Something like the following won't work:

```
with state.condition:
   cygrpc.decrement_active_thread_count()
   state.condition.wait()
   cygrpc.increment_active_thread_count()
```

Because fork could occur before the parent actually waits, and the child would inherit a locked condition variable on the call. (This could be kinda-sorta-okay, since the inherited call will anyways be failed in the child...but would prevent some of the normal API methods from functioning)

But yes, this behavior is non-obvious, nor is it obvious that 1 second is a good timeout value here (I suspect it could be too long, but then there are also performance impacts if its too short). I'll try to think of a cleaner way of doing this.

Reply:
I'm less worried about fork behavior, and more concerned about regular behavior with ```GRPC_ENABLE_FORK_SUPPORT=True```

Specifically, what happens if we hit the timeout before the send_message operation completes?  AFAICT, we would issue another write request.  It is illegal for the c-core to have more than one outstanding write requests, and would probably lead to a segfault.

Prior to this change, we had the invariant that if we pass the ```wait()``` call, an operation has completed.  We no longer have that invariant (here and below).

Reply:
Sounds good.  I'll let @mehrdada do the final approval, but I have no more concerns.

Reply:
We don't have to do it in this PR, but let's add a test where a slow service handler is processing the request so that we don't regress in the future.